### PR TITLE
Feature/ADF-905/Add properties repository, add hasProperty method to context

### PR DESCRIPTION
**Pull request by shpran**
# [ADF-905](https://oat-sa.atlassian.net/browse/ADF-905)

## Comanion PRs
* https://github.com/oat-sa/tao-core/pull/3356

### CODE REVIEW

File: core/Context/ContextInterface.php [hunk 1]
```diff
@@ -24,6 +24,8 @@
 
 interface ContextInterface
 {
+    public function hasParameter(string $parameter): bool;
```
Comment (gabrielfs7) [inline core/Context/ContextInterface.php:27]: - [ ] We need to make sure we are not breaking any context implementation

File: core/resource/Repository/PropertyRepository.php [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+use BadMethodCallException;
+use oat\generis\model\GenerisRdf;
+use oat\generis\model\OntologyRdf;
+use oat\search\helper\SupportedOperatorHelper;
+use oat\generis\model\Context\ContextInterface;
+use oat\generis\model\resource\Context\PropertyRepositoryContext;
+use oat\generis\model\resource\Contract\ResourceRepositoryInterface;
+use oat\generis\model\kernel\persistence\smoothsql\search\ComplexSearchService;
+
+class PropertyRepository implements ResourceRepositoryInterface
```
Comment (gabrielfs7) [inline core/resource/Repository/PropertyRepository.php:34]: - [ ] Lets add tests while QA is validating

**Review by gabrielfs7 — APPROVED** [review]

File: core/Context/ContextInterface.php (hunk 1 above)
Comment (shpran) [reply core/Context/ContextInterface.php:27]: I checked this one.
If everyone uses `AbstractContext` - this PR + `toa-core` PR will cover everything.
But I will everything again to be sure

**Review by hectoras — APPROVED** [review]

**Comment by AntonNeviarouski** [comment]
QA verified

**Pull request merged**